step-automation-packages/step-automation-packages-schema/src/main/java/step/automation/packages/yaml/schema/YamlAlertingRuleSchemaGenerator.java

```diff
@@ -0,0 +1,217 @@
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *
+ * This file is part of STEP
+ *
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.automation.packages.yaml.schema;
+
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObjectBuilder;
+import jakarta.json.spi.JsonProvider;
+import step.automation.packages.AutomationPackageNamedEntityUtils;
+import step.automation.packages.model.AutomationPackageAlertingRule;
+import step.automation.packages.yaml.rules.YamlConversionRule;
+import step.automation.packages.yaml.rules.YamlConversionRuleAddOn;
+import step.core.scanner.CachedAnnotationScanner;
+import step.core.yaml.schema.*;
+import step.handlers.javahandler.jsonschema.DefaultFieldMetadataExtractor;
+import step.handlers.javahandler.jsonschema.JsonSchemaCreator;
+import step.handlers.javahandler.jsonschema.JsonSchemaFieldProcessor;
+import step.handlers.javahandler.jsonschema.JsonSchemaPreparationException;
+import step.plugins.alerting.event.AlertingEventInterface;
+import step.plugins.alerting.rule.condition.AlertingRuleAction;
+import step.plugins.alerting.rule.condition.AlertingRuleCondition;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static step.core.scanner.Classes.newInstanceAs;
+
+public class YamlAlertingRuleSchemaGenerator {
```

Comment:
Do we need a custom schema generator for each model class? can't we do this in a generic way for "simple" model classes like AutomationPackageAlertingRule?

Reply:
@jeromecomte 
I also would like to apply some generic way to generate a json schema for any model class, but we have to solve the issue with inherited classes and `@JsonTypeInfo` (from my previous comment).